pkg/query/storeset.go

```diff
@@ -597,6 +630,105 @@ func (s *StoreSet) getActiveStores(ctx context.Context, stores map[string]*store
 			level.Warn(s.logger).Log("msg", "ignored rule store", "address", ruleAddr)
 		}
 	}
+
+	// Gather healthy stores map concurrently using info addresses. Build new store if does not exist already.
+	for _, infoSpec := range s.infoSpec() {
```

Comment:
All of this is baked in `storeset` but I would see it really part of endpoints

I would really just copy and add a new endpoint flow - it will be easier to main new thing than to try to understand later what to remove from mixed thing.

Then if someone wants to use endpoints store flag should be empty. WDYT?

Reply:
Agree! I will refactor this code and maybe create a new flow in a different `endpoint.go` file. WDYT ?

Reply:
SGTM